chain/arweave/src/adapter.rs

```diff
@@ -56,17 +62,35 @@ impl ArweaveTransactionFilter {
                     && !data_source.mapping.transaction_handlers.is_empty()
             })
             .map(|ds| {
-                base64_url::decode(&ds.source.owner.clone().unwrap_or_default()).unwrap_or_default()
+                let owner = ds.source.owner.clone().unwrap_or_default();
+
+                if owner.len() == HEX_SHA256_LEN {
+                    owner.into_bytes()
+                } else {
+                    base64_url::decode(&owner).unwrap_or_default()
+                }
             })
             .collect();
 
+        let (hex, long) = owners
+            .into_iter()
+            .partition::<Vec<Vec<u8>>, _>(|owner| owner.len() == HEX_SHA256_LEN);
+
+        let owners_hex: Vec<Vec<u8>> = hex
+            .into_iter()
+            .chain(long.iter().map(|bs| digest_bytes(bs).into_bytes()))
+            .collect();
+
+        let owners_pubkey = HashSet::from_iter(long);
         Self {
-            owners: HashSet::from_iter(owners),
+            match_all: owners_pubkey.contains(&vec![]),
```

Comment:
This only contains long form version no? This means it would match all if short form is provided.

Reply:
This was refactored in the meantime